src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/pullrequest/BitbucketPullRequestSCMHead.java

```diff
@@ -0,0 +1,57 @@
+package com.atlassian.bitbucket.jenkins.internal.scm.pullrequest;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketPullRequestState;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.mixin.ChangeRequestCheckoutStrategy;
+import jenkins.scm.api.mixin.ChangeRequestSCMHead2;
+
+import static java.util.Objects.requireNonNull;
+
+public class BitbucketPullRequestSCMHead extends SCMHead implements ChangeRequestSCMHead2 {
+
+    private static final String PR_ID_PREFIX = "pr#";
+
+    private final String branchName;
+    private final String latestCommit;
+    private final String pullRequestId;
+    private final BitbucketPullRequestState pullRequestState;
+    private final SCMHead target;
+
+    public BitbucketPullRequestSCMHead(String branchName, String latestCommit, String pullRequestId,
+                                       BitbucketPullRequestState pullRequestState, SCMHead target) {
+        super(PR_ID_PREFIX + pullRequestId);
```

Comment:
Jenkins (or something in the pipeline) actually uses `getName()` and not `getId()` and expects it to be the unique identifier for the PR build. This is why we have to do the refspec workaround [explained here](https://github.com/jaguilar-atl/atlassian-bitbucket-server-integration-plugin/pull/1/files#diff-933382a481d32a348aa5d3c47a2d3e7b63b23d1189c71f8f66f5d1ce98f0703dR50).